plugins/tickets.py

```diff
@@ -0,0 +1,1821 @@
+        self._prompt_task = asyncio.create_task(self._prompt())
+
+    async def ticket_updated(self, ticket):
+        """
+        Processes a ticket update.
+        """
+        if ticket.modid != self.modid:
+            # This should never happen
+            return
+
+        if not self.current_ticket:
+            # If we don't have a current ticket, this must be a new ticket
+            await self.deliver()
+        elif self.current_ticket.id == ticket.id:
+            if not self.current_ticket.delivered_id:
+                await self.deliver()
+            else:
+                self.current_ticket = ticket
+                # If the current ticket has been updated, update the current ticket message
+                args = {'embed': ticket.embed}
+                if ticket.stage == TicketStage.COMMENTED:
+                    args['content'] = None
+                await (await self.get_ticket_message()).edit(**args)
+
+    async def ticket_removed(self, ticket, reason=None):
+        """
+        Processes a removed ticket, with optional reason given.
+        """
+        if self.current_ticket and self.current_ticket.id == ticket.id:
+            # Post the reason
+            await self.user.send(reason or "Ticket #{} was removed from your queue!".format(ticket.id))
+
+            # Deliver next ticket
+            await self.deliver()
+
+    async def deliver(self):
+        """
+        Deliver the current ticket and refresh the prompt.
+        """
+        # TODO: Scheduling logic to handle delivery failure
+        # TODO: Logic to handle non-existent user
+        self.current_ticket = self.get_current_ticket()
+        if self.current_ticket:
+            logger.debug("Delivering ticket #{} to mod {}".format(self.current_ticket.id, self.modid))
+            try:
+                self._current_ticket_msg = await self.user.send(
+                    content="Please comment on the below ticket!",
+                    embed=self.current_ticket.embed
+                )
+            except discord.HTTPException:
+                # Reschedule
+                pass
+            else:
+                # Set current ticket to being delivered
+                self.current_ticket.update(stage=TicketStage.DELIVERED,
+                                           delivered_id=self._current_ticket_msg.id)
+
+                # Update the last prompt message
+                self.last_prompt_msgid = self._current_ticket_msg.id
+
+                # (Re-)schedule the next prompt update
+                await self.schedule_prompt()
+
+    async def process_message(self, message):
+        """
+        Process a non-command message from the moderator.
+        If there is a current active ticket, treat it as a comment.
+        Either way, update the last handled message in data.
+        """
+        if not priv.has_privilege('mod', message.author):
+            # Don't process messages from non-moderators at all.
+            return
```

Comment:
If they managed to trigger a ticket (TicketMod was created) I think they deserve to respond to it? Otherwise we'll just be hammering their DMs without a way for them to stop it?

Reply:
Fair, this seems more of an administrative decision.
What happens if a mod is demoted? Are they still allowed to finish their queue? (Before it is moved.)
We could also just not send reminders to non-mods.

Reply:
After a little consideration, definitely do that. I might rely on this behavior later.

Reply:
Removed the permission check in dd1db13, if the ticketmod exists they will now be able to respond to their own pending tickets.